tck/features/MatchAcceptance2.feature

```diff
@@ -0,0 +1,2023 @@
+      """
+      CREATE ()
+      """
+    When executing query:
+      """
+      MATCH p = (a)
+      RETURN p
+      """
+    Then the result should be:
+      | p    |
+      | <()> |
+    And no side effects
+
+  Scenario: Variable-length named path
+    Given an empty graph
+    And having executed:
+      """
+      CREATE ()
+      """
+    When executing query:
+      """
+      MATCH p = ()-[*0..]->()
+      RETURN p
+      """
+    Then the result should be:
+      | p    |
+      | <()> |
+    And no side effects
+
+  Scenario: Matching with aggregation
+    Given an empty graph
+    And having executed:
+      """
+      CREATE ({prop: 42})
+      """
+    When executing query:
+      """
+      MATCH (n)
+      RETURN n.prop AS n, count(n) AS count
+      """
+    Then the result should be:
+      | n  | count |
+      | 42 | 1     |
+    And no side effects
+
+  Scenario: Matching using already bound relationship
+    Given an empty graph
+    And having executed:
+      """
+      CREATE ()-[:T1]->(),
+             ()-[:T2]->()
+      """
+    When executing query:
+      """
+      MATCH ()-[r1]->()
+      WITH r1 AS r2
+      MATCH ()-[r2]->()
+      RETURN r2 AS rel
+      """
+    Then the result should be:
+      | rel   |
+      | [:T1] |
+      | [:T2] |
+    And no side effects
+
+  Scenario: Matching using already bound relationship with aggregation
+    Given an empty graph
+    And having executed:
+      """
+      CREATE ()-[:T1]->(),
+             ()-[:T2]->()
+      """
+    When executing query:
+      """
+      MATCH ()-[r1]->()
+      WITH r1 AS r2, count(*) AS c
+        ORDER BY c
+      MATCH ()-[r2]->()
+      RETURN r2 AS rel
+      """
+    Then the result should be:
+      | rel   |
+      | [:T1] |
+      | [:T2] |
+    And no side effects
+
+  Scenario: Matching using already bound relationship with aggregation and ORDER BY
+    Given an empty graph
+    And having executed:
+      """
+      CREATE ()-[:T1 {id: 0}]->(),
+             ()-[:T2 {id: 1}]->()
+      """
+    When executing query:
+      """
+      MATCH (a)-[r]->(b)
+      WITH a, r, b, count(*) AS c
+        ORDER BY c
+      MATCH (a)-[r]->(b)
+      RETURN r AS rel
+        ORDER BY rel.id
+      """
+    Then the result should be, in order:
+      | rel           |
+      | [:T1 {id: 0}] |
+      | [:T2 {id: 1}] |
+    And no side effects
+
+  Scenario: Matching with LIMIT and optionally matching using an already bound relationship
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (:A)-[:T]->(:B)
+      """
+    When executing query:
+      """
+      MATCH ()-[r]->()
+      WITH r
+        LIMIT 1
+      OPTIONAL MATCH (a2)-[r]->(b2)
+      RETURN a2, r, b2
+      """
+    Then the result should be:
+      | a2   | r    | b2   |
+      | (:A) | [:T] | (:B) |
+    And no side effects
+
+  Scenario: Matching with LIMIT and optionally matching using an already bound relationship and node
```

Comment:
"Matching with LIMIT and optionally matching using an already bound relationship and node" -> "Matching with LIMIT and optionally matching using a relationship and node that are both already bound"